akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/Http2FrameRendering.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2.framing
+
+import akka.event.Logging
+import akka.http.impl.engine.http2.Http2Protocol.FrameType.PUSH_PROMISE
+import akka.http.impl.engine.http2.Http2Protocol.SettingIdentifier
+import akka.http.impl.engine.http2._
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.util.ByteString
+
+import scala.collection.immutable
+
+/** INTERNAL API */
+private[http] class Http2FrameRendering extends GraphStage[FlowShape[FrameEvent, ByteString]] {
+
+  val frameIn = Inlet[FrameEvent]("Http2FrameRendering.frameIn")
+  val netOut = Outlet[ByteString]("Http2FrameRendering.netOut")
+
+  override def initialAttributes = Attributes.name(Logging.simpleName(getClass))
+
+  override val shape = FlowShape[FrameEvent, ByteString](frameIn, netOut)
+
+  override def createLogic(inheritedAttributes: Attributes) = new GraphStageLogic(shape) with InHandler with OutHandler {
+    setHandlers(frameIn, netOut, this)
+
+    // TODO remove
+    object settings {
+      private var _outMaxFrameSize: Int = 16384 // default
```

Comment:
Here we need these settings but they need to be changed when we send out the SettingAck. I think that might now have been lost somewhere?